Added more test coverage for the portmapper package

mavenugo opened the pull request:
- Removed MockProxy
- TestMapAllPortsSingleInterface is greedy and consumes a lot of time and resources and hence introduced the -test.short for all other tests except this test.
- Added UDP test
- explicit test for execProxy to satisfy code-coverage tools

Signed-off-by: Madhu Venugopal <madhu@docker.com>

aboch commented on portmapper/mapper_test.go:
```diff
@@ -77,6 +84,7 @@ func TestMapPorts(t *testing.T) {
 }
 
 func TestGetUDPKey(t *testing.T) {
+	defer netutils.SetupTestNetNS(t)()
```
This is not needed.

aboch commented on portmapper/mapper_test.go:
```diff
@@ -87,6 +95,7 @@ func TestGetUDPKey(t *testing.T) {
 }
 
 func TestGetTCPKey(t *testing.T) {
+	defer netutils.SetupTestNetNS(t)()
```
Not needed

aboch commented on portmapper/mapper_test.go:
```diff
@@ -97,6 +106,7 @@ func TestGetTCPKey(t *testing.T) {
 }
 
 func TestGetUDPIPAndPort(t *testing.T) {
+	defer netutils.SetupTestNetNS(t)()
```
Not needed

aboch commented:
Checked out your changes and run the plain go test (sudo -E go test) in portmapper pkg.
Go test does not complete and my machine CPU get stuck in some kernel process.

Can you please check.

mavenugo commented:
@aboch as I mentioned in the PR comments, the test `TestMapAllPortsSingleInterface` is greedy. Can you use `-test.short` flag to confirm ?

I will try to change this to use a custom flag.

aboch commented:
Tried with short flag, now it's fast.
I still get the failure, but might be unrelated. At least it is not related to the long test.

$ sudo -E go test  -test.short=true
INFO[0000] Stopping proxy on tcp/192.168.0.1:9999 for tcp/172.168.1.1:8888 (accept tcp 192.168.0.1:9999: bad file descriptor) 
--- FAIL: TestExecProxy (0.00s)
	mapper_test.go:224: execProxy is not supposed to exit
FAIL
exit status 1
FAIL	github.com/docker/libnetwork/portmapper	0.133s


Agree, I think a specific test flag, if possible, would be the right approach.

mavenugo commented:
@aboch Thanks. Can you please try it again ? I removed the `short` and replaced with custom flag and is in an opt-in & hence you can just run `go test` without any additional flags.

aboch commented:
Now it works thanks.
The error I was facing was probably due to the fact I interrupted the previous long run of the test.

mavenugo commented:
@aboch @mrjana Fixed the code to bring in a balance between code-coverage and not-having-to-wait-for-30-mins :-) . Instead of skipping the `TestMapAllPortsSingleInterface` test, we use the Mock proxy for this case alone.

mrjana commented:
LGTM
